telegraf.js

```diff
@@ -108,7 +108,10 @@ class Telegraf extends Composer {
           const { timeout, limit, allowedUpdates, stopCallback } = config.polling || {}
           return this.telegram.deleteWebhook()
             .then(() => this.startPolling(timeout, limit, allowedUpdates, stopCallback))
-            .then(() => debug('Bot started with long-polling'))
+            .then(() => {
+              debug('Bot started with long-polling')
+              return { ok: true, error: null, type: 'long-polling' }
```

Comment:
What will you do with that object then?
How is it solving your issue?

Reply:
several options for starting the bot are possible
for example, in any case, the Object will be returned
in case of failure
```js
return { ok: false, error: { message: 'Launch failed', code: 'ERRORCODE' } }
```
in case of success
```js
return { ok: true, error: null, type: 'long-polling'  }
```
or
```js
return { ok: true, error: null, type: 'webhook'  }
```

**polling**
https://github.com/telegraf/telegraf/blob/b1e19eca8905303e9937c4f791b8d9d7b1f946ca/telegraf.js#L107-L112
to
```js
        if (!config.webhook) {
          const { timeout, limit, allowedUpdates, stopCallback } = config.polling || {}
          return this.telegram.deleteWebhook()
            .then(() => this.startPolling(timeout, limit, allowedUpdates, stopCallback))
            .then(() => {
              debug('Bot started with long-polling')
              return { ok: true, error: null, type: 'long-polling' }
            })
        }
```

**webhook**
https://github.com/telegraf/telegraf/blob/b1e19eca8905303e9937c4f791b8d9d7b1f946ca/telegraf.js#L127-L129
to
```js
        return this.telegram
          .setWebhook(`https://${domain}${hookPath}`)
          .then(() => {
            debug(`Bot started with webhook @ https://${domain}`)
            return { ok: true, error: null, type: 'webhook' }
          })
```

**and catching error**

https://github.com/telegraf/telegraf/blob/b1e19eca8905303e9937c4f791b8d9d7b1f946ca/telegraf.js#L131-L134
to
```js
      .catch((err) => {
        console.error('Launch failed')
        console.error(err.stack || err.toString())
        return { ok: false, error: { message: 'Launch failed', code: err.code ? err.code : null } }
      })
```

Reply:
When launching a bot through a 'launch' fails/success, I cannot track this. I tried different methods. the only remaining way is to make changes and return an object that says success or error.

Example. Telegram servers are blocked in Russia. Running a bot without a proxy fails with the 'ECONNRESET' error. if the proxy does not work, then the start fails with the error 'ECONNREFUSED'. If there is no network, then the start fails with the 'ENOTFOUND' error.

And I can’t catch it because nothing is returning.

After making my changes, I can get an object that says whether the launch was successful or not. and what error was the cause.

```js
bot.launch(options = {})
     .then((data) => {
       if (data !== undefined && !data.ok && data.error !== null)
         console.log ('The launch was a error. message: %s, code: %s',  data.error.message, data.error.code);
       if (data !== undefined && data.ok)
         console.log ('The launch was a success.');
     });
```